src/integrationtest/java/com/ericsson/gerrit/plugins/eiffel/integrationtest/ServiceIntegrationSteps.java

```diff
@@ -0,0 +1,282 @@
+
+    @Given("^the project is configured to send eiffel events with publish url \"([^\"]*)\"$")
+    public void theProjectIsConfiguredToSendEiffelEventsWithPublishUrl(String publishUrl)
+            throws ClientProtocolException, URISyntaxException, IOException {
+        ResponseEntity response = enableEiffelMessaging(publishUrl);
+        int expected = 200;
+        int actual = response.getStatusCode();
+        assertEquals("Failed to configure project", expected, actual);
+    }
+
+    @Given("^a change is created$")
+    public void aChangeIsSubmitted()
+            throws ClientProtocolException, URISyntaxException, IOException {
+        ResponseEntity response = submitChange();
+        int expected = 201;
+        int actual = response.getStatusCode();
+        assertEquals("Change is submitted", expected, actual);
+    }
+
+    @Then("^I should find an eiffel event in rabbitmq$")
+    public void iShouldFindAnEiffelEventInRabbitmq() throws IOException, TimeoutException {
+        List<String> messages = consumeMessages(1, 10000);
+        assertEquals(1, messages.size());
+    }
+
+    private void generateGerritXAuthToken() throws Exception {
+        try {
+            HttpRequest request = new HttpRequest(HttpMethod.GET);
+            ResponseEntity response;
+            response = request.setBaseUrl(GERRIT_BASE_URL).setHeader("Cookie", gerritAccountCookie)
+                    .setHeader("Referer", "http://localhost:8080/login").setEndpoint("/")
+                    .performRequest();
+
+            Header[] headers = response.getHeaders();
+            Header gerritAccountHeader = headers[1];
+
+            String gerritXAuthHeader = gerritAccountHeader.getValue();
+            String regexp = "XSRF_TOKEN=(.+);";
+            gerritXAuthToken = getFirstMatchGroupFromText(gerritXAuthHeader, regexp);
+        } catch (URISyntaxException e) {
+            String stacktrace = ExceptionUtils.getStackTrace(e);
+            throw new URISyntaxException("Failed to generate gerritXAuthToken", stacktrace);
+        } catch (IOException e) {
+            throw new IOException("Failed to generate gerritXAuthToken", e);
+        }
+    }
+
+    private void generateGerritAccountCookie() throws Exception {
+        try {
+            CloseableHttpClient httpClientNoRedirect = HttpClientBuilder.create()
+                    .disableRedirectHandling().build();
+            HttpExecutor httpExecutor = new HttpExecutor(httpClientNoRedirect);
+            HttpRequest requestLogin = new HttpRequest(HttpMethod.GET, httpExecutor);
+
+            ResponseEntity response = requestLogin.setBaseUrl(GERRIT_BASE_URL)
+                    .setEndpoint("/login/?account_id=1000000")
+                    .addParameter("http.protocol.handle-redirects", "false").performRequest();
+
+            Header[] headers = response.getHeaders();
+            Header gerritAccountHeader = headers[4];
+            gerritAccountCookie = gerritAccountHeader.getValue();
```

Comment:
please add this. before gerritAccountCookie. Do this to all global vars when you set them since it is not clear.